### PR TITLE
Avoid play store warnings when building libraries

### PR DESCRIPTION
**Pull request by dotMorten**
When you compile a library you might not want to target the latest SDK, as it could hose users targeting lower-end api-levels. However you currently get a warning that really only applies to applications, and not class libraries.
This PR skips the API Level check when the target isn't an application.
Also there were two near-identical warnings on the same check, so I consolidated those into a single warning (it looks like this was historical as updates vs new apps were on different timelines)

@jonathanpeppers

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
build

**Comment by dellis1972** [comment]
We will also need to update https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/messages/README.md to remove the deleted warning as well.

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs [hunk 1]
```diff
@@ -188,6 +188,7 @@ public void UseLatestAndroidSdk (string buildtools, string jdk, ApiInfo[] apis,
 				UseLatestAndroidPlatformSdk = useLatestAndroidSdk,
 				AotAssemblies = false,
 				SequencePointsMode = "None",
+				AndroidApplication = true,
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs]: Are these changes needed? given that the default for `ResolveAndroidTooling.AndroidApplication` is true?

File: src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs [hunk 2]
```diff
@@ -185,11 +187,9 @@ public override bool Execute ()
 			}
 
 			int apiLevel;
-			if (int.TryParse (AndroidApiLevel, out apiLevel)) {
-				if (apiLevel < 26)
-					Log.LogCodedWarning ("XA0113", $"Google Play requires that new applications must use a TargetFrameworkVersion of v8.0 (API level 26) or above. You are currently targeting {TargetFrameworkVersion} (API level {AndroidApiLevel}).");
+			if (AndroidApplication && int.TryParse (AndroidApiLevel, out apiLevel)) {
 				if (apiLevel < 26)
-					Log.LogCodedWarning ("XA0114", $"Google Play requires that application updates must use a TargetFrameworkVersion of v8.0 (API level 26) or above. You are currently targeting {TargetFrameworkVersion} (API level {AndroidApiLevel}).");
+					Log.LogCodedWarning ("XA0113", $"Google Play requires that new applications and updates must use a TargetFrameworkVersion of v8.0 (API level 26) or above. You are currently targeting {TargetFrameworkVersion} (API level {AndroidApiLevel}).");
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs]: We will need to update the documentation to reflect the change in wording, and probably delete the documentation for `XA0114` if it is no longer required.

Comment (jonpryor) [reply src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs]: Relevant documentation files which would need to be updated are:

  * https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/messages/README.md
  * https://github.com/xamarin/xamarin-android/blob/master/Documentation/guides/messages/xa0113.md

Comment (dotMorten) [reply src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs]: @dellis1972 If you delete the doc, what happens if users are using the older tooling and can't get info on the error?
Do you have people who are good at writing doc for stuff like this? It goes a bit beyond what I intended

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs (hunk 1 above)
Comment (dotMorten) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ResolveSdksTaskTests.cs]: Technically not - but this matches what is happening in the .targets file where it always gets set. I made true default, just in case someone did what this test is doing manually so it wouldn't be breaking.

File: src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs (hunk 2 above)
Comment (jonpryor) [reply src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs]: > If you delete the doc, what happens if users are using the older tooling and can't get info on the error?

I don't understand the question.

Additionally, at present most of the docs in this repo are "advisory," to the extent that they're not on `docs.microsoft.com`.  (We intend to fix this, but it hasn't happened yet.)

> It goes a bit beyond what I intended.

I believe Dean was only requesting that the existing docs that we have -- which I linked to [above](https://github.com/xamarin/xamarin-android/pull/2754#discussion_r259022091) -- be updated to use the same message you altered in this PR, for consistency.  That is all.

Comment (dotMorten) [reply src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs]: Regarding deleting doc... It's still possible for users to get XA0114 as long as people are on older versions of the build task. If I delete the doc, people looking for information on that bug will immediately be unsuccessful once this PR merges.

I'll fix the message in XA0113...

**Comment by jonpryor** [comment]
@monojenkins build
